app/controllers/address_controller.rb

```diff
@@ -0,0 +1,35 @@
+class AddressController < ApplicationController
+  def create
+    user = User.find_by(verification_token: params[:verification_token])
+    address = user.addresses.build(permitted_address_params)
+
+    if address.save
+      order = logged_in_user_cart_order
+      order.shipping_address_id = address.id
+      order.save
```

Comment:
what happens if order's save fails?

Reply:
Add an if-else block which on order save failure redirects to the shipping address setting form page with a flash message that "Unable to save shipping address for the order"